src/cpp/rtps/history/WriterHistory.cpp

```diff
@@ -94,13 +95,14 @@ bool WriterHistory::add_change_(
     ++m_lastCacheChangeSeqNum;
     a_change->sequenceNumber = m_lastCacheChangeSeqNum;
     Time_t::now(a_change->sourceTimestamp);
-    a_change->num_sent_submessages = 0;
+    a_change->writer_info.num_sent_submessages = 0;
```

Comment:
The initialization is done in RTPSWriter::new_change(), before calling this function. This line was modified only because it is already there, but I think the other two fields are already initializated.